contracts/StakeHub.sol

```diff
@@ -1011,6 +1046,138 @@ contract StakeHub is SystemV2, Initializable, Protectable {
         }
     }
 
+     /**
+     * @notice Adds a new NodeID (of type bytes32) to the validator's registry.
+     * @param validator The address of the validator.
+     * @param newNodeID The NodeID to be added (must be nonzero).
+     */
+    function addNodeID(
+        address validator,
+        bytes32 newNodeID
+    ) external whenNotPaused notInBlackList validatorExist(validator) onlyAuthorized(validator) {
+        require(newNodeID != bytes32(0), "Invalid NodeID");
+        require(
+            validatorNodeIDs[validator].length < maxNodeIDs,
+            "Maximum NodeIDs reached"
+        );
+        
+        // Check for duplicate NodeID
+        for (uint256 i = 0; i < validatorNodeIDs[validator].length; i++) {
+            require(
+                validatorNodeIDs[validator][i] != newNodeID,
+                "NodeID already exists"
+            );
+        }
+
+        validatorNodeIDs[validator].push(newNodeID);
+        emit NodeIDAdded(validator, newNodeID);
+    }
+
+    /**
+     * @notice Removes a specified NodeID from the validator's registry.
+     * @param validator The address of the validator.
+     * @param targetNodeID The NodeID to be removed.
+     */
+    function removeNodeID(
+        address validator,
+        bytes32 targetNodeID
+    ) external whenNotPaused notInBlackList validatorExist(validator) onlyAuthorized(validator) {
+        uint256 length = validatorNodeIDs[validator].length;
+        require(length > 0, "No NodeIDs registered for validator");
+
+        // Find the index of the target NodeID
+        uint256 indexToRemove = length; // default invalid index
+        for (uint256 i = 0; i < length; i++) {
+            if (validatorNodeIDs[validator][i] == targetNodeID) {
+                indexToRemove = i;
+                break;
+            }
+        }
+        require(indexToRemove < length, "NodeID not found for validator");
+
+        // Remove the NodeID using the swap-and-pop technique
+        if (indexToRemove != length - 1) {
+            validatorNodeIDs[validator][indexToRemove] = validatorNodeIDs[validator][length - 1];
+        }
+        validatorNodeIDs[validator].pop();
+        emit NodeIDRemoved(validator, targetNodeID);
+
+        // Clear the array if empty to save storage
+        if (validatorNodeIDs[validator].length == 0) {
+            delete validatorNodeIDs[validator];
+        }
+    }
+
+    /**
+     * @notice Replaces an existing NodeID with a new one for a validator.
+     * @param validator The address of the validator.
+     * @param oldNodeID The NodeID to be replaced.
+     * @param newNodeID The new NodeID to replace the old one with.
+     */
+    function replaceNodeID(
```

Comment:
I think the `replaceNodeIDs` logic could be very easy.  input the new adding NodeIDs, and just delete the old NodeIDs, and assigning the new ones.